java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -236,18 +237,14 @@ public MantaClient(final ConfigContext config) {
     }
 
     /**
-     * Creates a new instance of the Manta client with the specified connection factory.
-     * This method is typically used by unit tests to inject connection factory
-     * customizations so we can simulate different network events.
+     * Creates a new instance of the Manta client based on user-provided connection objects. This allows for a larger
+     * degree of customization at the cost of
      *
      * @param config The configuration context that provides all of the configuration values
-     * @param keyPair cryptographic signing key pair used for HTTP signatures
-     * @param connectionFactory connection factory instance used to tune connection settings
+     * @param connectionFactoryConfigurator pre-configured objects for use with a MantaConnectionFactory
      */
-    MantaClient(final ConfigContext config,
-                final KeyPair keyPair,
-                final MantaConnectionFactory connectionFactory,
-                final ThreadLocalSigner signer) {
```

Comment:
Dropping the ability to pass in a signer is intentional, correct?

Reply:
This constructor was only used in a single test which was has been disabled for a while (by being commented out in the `testng.xml` file) and needs to be revisted. That test case is documented in #354 and could be achieved with the `MantaConnectionFactoryConfigurator`  and some simple mocks.